### PR TITLE
CIRC-1342 implement fetchRequests method

### PR DESCRIPTION
**Pull request by viktor-draban**
Link to [ticket](https://issues.folio.org/browse/CIRC-1342)

there was added implementation for fetchRequests method

### CODE REVIEW

File: src/main/java/org/folio/circulation/services/support/ItemsInTransitReportContext.java [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 import java.util.Map;
 
+import lombok.AllArgsConstructor;
```
Comment (sherzodn) [inline src/main/java/org/folio/circulation/services/support/ItemsInTransitReportContext.java:5]: Please use all Lombok imports in one place

File: src/test/java/org/folio/circulation/services/ItemsInTransitReportServiceTest.java [hunk 2]
```diff
@@ -1,23 +1,30 @@
...
 import org.folio.circulation.support.results.Result;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
 
-import io.vertx.core.json.JsonObject;
+import java.util.concurrent.CompletableFuture;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
```
Comment (sherzodn) [inline src/test/java/org/folio/circulation/services/ItemsInTransitReportServiceTest.java:13]: according to convention, static imports' orders should come before non-static imports

File: src/main/java/org/folio/circulation/services/support/ItemsInTransitReportContext.java (hunk 1 above)
Comment (viktor-draban) [reply src/main/java/org/folio/circulation/services/support/ItemsInTransitReportContext.java]: done

File: src/test/java/org/folio/circulation/services/ItemsInTransitReportServiceTest.java (hunk 2 above)
Comment (viktor-draban) [reply src/test/java/org/folio/circulation/services/ItemsInTransitReportServiceTest.java]: thanks

File: src/main/java/org/folio/circulation/infrastructure/storage/requests/RequestRepository.java [hunk 3]
```diff
@@ -143,6 +146,14 @@ public static RequestRepository using(org.folio.circulation.support.Clients clie
       .fetch(id);
   }
 
+  public CompletableFuture<Result<MultipleRecords<Request>>> findOpenStatusRequestsBy(
```
Comment (alexanderkurash) [inline src/main/java/org/folio/circulation/infrastructure/storage/requests/RequestRepository.java:149]: -> findOpenRequests

File: src/main/java/org/folio/circulation/services/ItemsInTransitReportService.java [hunk 4]
```diff
@@ -120,9 +122,11 @@
   }
 
   private CompletableFuture<Result<ItemsInTransitReportContext>> fetchRequests(
-    Result<ItemsInTransitReportContext> context) {
+    ItemsInTransitReportContext context) {
 
-    return completedFuture(context);
+    return requestRepository.findOpenRequests(context.getItems().keySet())
+      .thenApply(r -> r.map(records -> toMap(records.getRecords(), Request::getId)))
```
Comment (roman-barannyk) [inline src/main/java/org/folio/circulation/services/ItemsInTransitReportService.java:128]: mapResult?

File: src/main/java/org/folio/circulation/infrastructure/storage/requests/RequestRepository.java [hunk 5]
```diff
@@ -143,6 +146,14 @@ public static RequestRepository using(org.folio.circulation.support.Clients clie
       .fetch(id);
   }
 
+  public CompletableFuture<Result<MultipleRecords<Request>>> findOpenRequests(
```
Comment (roman-barannyk) [inline src/main/java/org/folio/circulation/infrastructure/storage/requests/RequestRepository.java:149]:
```suggestion
  public CompletableFuture<Result<MultipleRecords<Request>>> findOpenRequestsByItemIds(
```

Comment (viktor-draban) [reply src/main/java/org/folio/circulation/infrastructure/storage/requests/RequestRepository.java:149]: image

File: src/main/java/org/folio/circulation/services/ItemsInTransitReportService.java (hunk 4 above)
Comment (viktor-draban) [reply src/main/java/org/folio/circulation/services/ItemsInTransitReportService.java:128]: why ?

File: src/main/java/org/folio/circulation/infrastructure/storage/requests/RequestRepository.java (hunk 5 above)
Comment (viktor-draban) [reply src/main/java/org/folio/circulation/infrastructure/storage/requests/RequestRepository.java]: done

File: src/main/java/org/folio/circulation/services/ItemsInTransitReportService.java (hunk 4 above)
Comment (viktor-draban) [reply src/main/java/org/folio/circulation/services/ItemsInTransitReportService.java]: done

File: src/main/java/org/folio/circulation/services/ItemsInTransitReportService.java [hunk 6]
```diff
@@ -120,9 +125,12 @@
...
 
   private CompletableFuture<Result<ItemsInTransitReportContext>> fetchRequests(
-    Result<ItemsInTransitReportContext> context) {
+    ItemsInTransitReportContext context) {
 
-    return completedFuture(context);
+    return requestRepository.findOpenRequestsByItemIds(context.getItems().keySet())
+      .thenApply(mapResult(this::toList))
+      .thenApply(mapResult(requests -> toMap(requests, Request::getId)))
+      .thenApply(r -> r.map(context::withRequests));
```
Comment (roman-barannyk) [inline src/main/java/org/folio/circulation/services/ItemsInTransitReportService.java:133]:
```suggestion
      .thenApply(mapResult(context::withRequests));
```

Comment (viktor-draban) [reply src/main/java/org/folio/circulation/services/ItemsInTransitReportService.java]: thanks

File: src/main/java/org/folio/circulation/infrastructure/storage/requests/RequestRepository.java [hunk 7]
```diff
@@ -143,6 +146,14 @@ public static RequestRepository using(org.folio.circulation.support.Clients clie
       .fetch(id);
   }
 
+  public CompletableFuture<Result<MultipleRecords<Request>>> findOpenRequestsByItemIds(
+    Collection<String> itemIds) {
+
+    return findWithMultipleCqlIndexValues(requestsStorageClient,
+      "requests", Request::from)
+      .findByIdIndexAndQuery(itemIds, "itemId", exactMatchAny("status", openStates()));
```
Comment (roman-barannyk) [inline src/main/java/org/folio/circulation/infrastructure/storage/requests/RequestRepository.java:154]:
```suggestion
    return findWithMultipleCqlIndexValues(requestsStorageClient, "requests", Request::from)
      .findByIdIndexAndQuery(itemIds, "itemId", exactMatchAny("status", openStates()));
```

Comment (viktor-draban) [reply src/main/java/org/folio/circulation/infrastructure/storage/requests/RequestRepository.java]: done

**Review by roman-barannyk — APPROVED** [review]

**Pull request merged**